modules/dhcp_common/subnet_service.rb

```diff
@@ -115,10 +117,16 @@ def find_lease_by_ip(subnet_address, ip_address)
       m.synchronize { leases_by_ip[subnet_address, ip_address] }
     end
 
-    def find_host_by_ip(subnet_address, ip_address)
+    def find_hosts_by_ip(subnet_address, ip_address)
       m.synchronize { reservations_by_ip[subnet_address, ip_address] }
     end
 
+    def find_host_by_ip(subnet_address, ip_address)
```

Comment:
You kept this for compatibility reasons, correct? Only isc dhcpd and libvirt providers use subnet service, I think updating both to use "find_hosts_by_ip" method is a better/safer approach: this changes internal model and I'm paranoid that some code will rely on one record per ip assumption. I'd much rather have an "undefined method" failure than something (potentially) subtle.